build.sbt

```diff
@@ -132,18 +126,15 @@ lazy val commonSettings = Def.settings(
 
 lazy val root = project
   .in(file("."))
-  .enablePlugins(PlayRootProject, ScalaJSPlugin)
+  .enablePlugins(PlayRootProject, ScalaJSPlugin, PlayNoPublish)
```

Comment:
(I'll ignore `ScalaJSPlugin`)

The `enablePlugins` above can be expanded to:

1. `PlayRootProject` expands to `PlayRootProjectBase &&! PlaySbtCompat.optScriptedAutoPlugin`
1. `PlayNoPublish` expands to `PlayNoPublishBase &&! BintrayPlugin &&! Sonatype`

Then we have:

```
enablePlugins(
  PlayRootProjectBase &&! PlaySbtCompat.optScriptedAutoPlugin, // exclude(ScriptedAutoPlugin)
  PlayNoPublishBase &&! BintrayPlugin &&! Sonatype // exclude(BintrayPlugin, Sonatype)
)
```

Now, `PlayRootProjectBase` brings some stuff transitively:

```
object PlayRootProjectBase extends AutoPlugin {
  override def trigger = noTrigger
  override def requires = PlayBuildBase && PlayBintrayBase && PlaySonatypeBase && PlayReleaseBase
```

So the end result looks like:

```
enablePlugins(
  PlayRootProjectBase &&! PlaySbtCompat.optScriptedAutoPlugin, // exclude(ScriptedAutoPlugin)
  PlayBuildBase, 
  PlayBintrayBase,  // brings `BintrayPlugin` transitively
  PlaySonatypeBase, // brings `Sonatype` transitively
  PlayReleaseBase,

  PlayNoPublishBase &&! BintrayPlugin &&! Sonatype // exclude(BintrayPlugin, Sonatype)
  PlayBuildBase,    // transitively added by `PlayNoPublishBase`
)
```

Which, I guess, looks like:

```
enablePlugins(
  PlayRootProjectBase
  PlayBuildBase, 
  PlayReleaseBase,
  PlayNoPublishBase
)
```

- - - - - - - - - - - 

IIUC, the purpose is to not publish the `root` project. In that case, we don't want `PlayReleaseBase`. 

Then, we could be tempted to also remove `PlayRootProjectBase` (which is the one bringing in `PlayReleaseBase` but `PlayRootProjectBase` is defining the [`crossScalaVersions`](https://github.com/playframework/interplay/blob/master/src/main/scala/interplay/PlayBuildBase.scala#L356-L362))

I don't know what I'm talking about anymore.

 - - - - - - - - - - 

OMG, this is so confusing.

According to [the docs](https://github.com/playframework/interplay#the-root-project) this aggregating `root` project should be a `PlayRootProject` and then, each of the aggregated projects that must not be published should be `PlayNoPublish`.

Ofc, `interplay` being created before sbt `1.x` it's possible that some assumptions no longer hold true in the sbt `1.3+` ecosystem.

Reply:
I think the `NoPublish` is unnecessary and your change above:

```

  publishTo := Some(
    if (isSnapshot.value)
      Opts.resolver.sonatypeSnapshots
    else
      Opts.resolver.sonatypeStaging
  ),
```

is enough.